scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryIO.scala

```diff
@@ -247,6 +263,35 @@ object BigQueryTypedTable {
     ): WriteParam = apply(s, wd, cd, td, tp, DefaultExtendedErrorInfo)(defaultInsertErrorTransform)
   }
 
+  private[this] def tableRow(table: Table): BigQueryTypedTable[TableRow] =
+    BigQueryTypedTable(
+      beam.BigQueryIO.readTableRows(),
+      beam.BigQueryIO.writeTableRows(),
+      table,
+      BigQueryUtils.convertGenericRecordToTableRow(_, _)
+    )
+
+  private[this] def genericRecord(table: Table): BigQueryTypedTable[GenericRecord] =
+    BigQueryTypedTable(
+      _.getRecord(),
+      identity[GenericRecord],
+      (genericRecord: GenericRecord, _: TableSchema) => genericRecord,
+      table
+    )
+
+  /**
+   * Creates a new instance of [[BigQueryTypedTable]] based on the supplied [[Format]].
+   *
+   * NOTE: LogicalType support when using `Format.GenericRecord` has some caveats:
+   * Reading: Bigquery types DATE, TIME, DATIME will be read as STRING
+   * Writting: Supports LogicalTypes only for DATE and TIME.
+   *           DATETIME is not yet supported. https://issuetracker.google.com/issues/140681683
+   */
+  def apply(table: Table, format: Format): BigQueryTypedTable[format.F] = format match {
```

Comment:
Could we not just expose the 2 methods and skip the `Format` trait entirely ?
```suggestion
  def genericRecord(table: Table): BigQueryTypedTable[GenericRecord] = ???
  def tablerow(table: Table): BigQueryTypedTable[TableRow] = ???
```